index.ts

```diff
@@ -229,11 +241,11 @@ wss.on("request", async function (request) {
 
     switch (data.type) {
       case "join":
-        room.connections.push({name, connection, id});
+        room.connections.push({name, connection, id, readOnly});
         msg = JSON.stringify({
           type: "connected",
           connected: room.connections.reduce((rv, cv) => {
-            if (!rv.find((item) => item.id == cv.id)) {
+            if (!cv.readOnly && !rv.find((item) => item.id == cv.id)) {
```

Comment:
This will still send an update about a new participant to all connected clients of that room although we filter out the `readOnly` client here and thus there is no change to be sent?

Reply:
This is not sent to all connected clients but rather to the freshly connected client to populate his user list. view only connections should still be able to see connected participants.